contracts/PowerBank.sol

```diff
@@ -0,0 +1,116 @@
+pragma solidity 0.7.4;
+
+/* solhint-disable indent */
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155Holder.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+import "./RuleManager.sol";
+
+// TODO: consider  order of contract  deployment. May not want to deploy all together
+// this will impact the modifiers that are important for securiy
+// for not deploying supporting contracts as part of main contract
+
+// TODO: custom implementation of erc1155
+// enable defining your own podId
+// enable transfer of the podId token
+// only allow for one token per user
+
+contract PowerToken is ERC1155 {
+
+    constructor() ERC1155("POD") {}
+
+    //TODO only power bank
+    function mint(address _to, uint256 _id, uint256 _amount, bytes memory _data) public {
+        _mint(_to,_id,_amount,_data);
+    }
+
+}
+
+contract PowerBank is ERC1155Holder {
+    using SafeMath for uint256;
+    using Address for address;
+
+    PowerToken public powerToken;
+    // Maps podIds to pods
+    mapping(uint256 => Pod) pods;
+
+    // TODO: Need a way to iterate through both pods and users in the pods.
+    struct Pod {
+        address creator;
+        uint256 totalSupply;
+    }
+
+    // probably a better way to manage  this
+    // dependent on how we are managing contract deployment
+    modifier onlyProtocol {
+        // require(
+        //     // TODO: Should these be the same modifier?
```

Comment:
Any reason these are turned off? Is it testing purposes or were they just never working?

Reply:
This is addressed in the controller PR